server/client.go

```diff
@@ -917,13 +920,21 @@ func (c *client) flushOutbound() bool {
 		}
 	}
 
+	// Check that if there is still data to send and writeLoop is in wait,
+	// we need to signal
+	if c.out.pb > 0 && c.out.sgw {
+		c.out.sg.Signal()
+	}
+
 	return true
 }
 
 // flushSignal will use server to queue the flush IO operation to a pool of flushers.
 // Lock must be held.
 func (c *client) flushSignal() {
-	c.out.sg.Signal()
+	if c.out.sgw {
```

Comment:
Do we think this is saving us much?

Reply:
Did a quick and dirty benchmark of simulating signaling a routine going into a wait, with and without a check with boolean:
```
Benchmark_SignalWithoutCheck-8   	50000000	        41.6 ns/op
Benchmark____SignalWithCheck-8   	100000000	        19.9 ns/op
```
But more importantly, even if we don't gain much, why not checking? Signaling while not in wait is just wasted because a signal while not in wait is lost anyway. Since we use mutex to protect wait()/signal(), it makes sense to be to know if we are in wait or not.

Reply:
ok that is convincing!